lib/google_places/spot.rb

```diff
@@ -426,9 +431,9 @@ def initialize(json_result_object, api_key)
       @reference                  = json_result_object['reference']
       @place_id                   = json_result_object['place_id']
       @vicinity                   = json_result_object['vicinity']
-      @lat                        = json_result_object['geometry']['location']['lat']
-      @lng                        = json_result_object['geometry']['location']['lng']
-      @viewport                   = json_result_object['geometry']['viewport']
+      @lat                        = json_result_object&.[](:geometry)&.[](:location)&.[](:lat)
```

Comment:
Afraid `hash&.[](:a)` syntax was also introduced in Ruby 2.3 😢

Is this change because we can't now guarantee that `geometry` is set within the results hash?

If `geometry` is set, can we assume that all the other values *are* set - as we could just do something like:
```ruby
@lat = json_result_object[:geometry] ? json_result_object['geometry']['location']['lat'] : nil
```

Reply:
@edwinwills - sorry for not getting back quicker. yes, i see i need to use some other syntax there still.  as you mentioned, the change is because we can't guarantee any field has been requested when specific *fields* are passed in the request. the ternary looks good. i'll have the changes in soon with that or similar.

Reply:
No worries, and thanks so much for your efforts 😄